src/algorithms/bfs.jl

```diff
@@ -6,24 +6,22 @@ which is sequential and runs through every vertex in the graph. If you know
 the maximum number of active vertices, call `bfs(s, source, nv)` which is faster.
 """
 function bfs(s::Stinger, source::Int64)
-    nv = get_nv(s)
-    bfs(s, source, nv)
+    bfs(s, source, s[max_nv])
 end
 
 function bfs(s::Stinger, source::Int64, nv::Int64)
-    nv>source || throw(DimensionMismatch("Attempting to run BFS with source $source in a graph with only $nv vertices."))
+    #nv>source || throw(DimensionMismatch("Attempting to run BFS with source $source in a graph with only $nv vertices."))
```

Comment:
Adding 
```julia
if source>nv
        return zeros(Int64, 0)
end
```

also results in type instabilities.

One option would be to separate the `bfs` logic into a different function (`bfskernel`), perform the check in the `bfs` function and then call the `bfskernel` function from `bfs`.

```julia
function bfs(s::Stinger, source::Int64, nv::Int64)
    #nv>source || throw(DimensionMismatch("Attempting to run BFS with source $source in a graph with only $nv vertices."))
    if source>nv
        return zeros(Int64, 0)
    end
    bfskernel(s, source, nv)
end
```
The `DimensionMismatch` creates a `Union` in `@code_warntype`. I'm not sure if that impacts performance. The empty array option does not have such an issue.

What do you think is the better option @jpfairbanks ?